File: army_recruitment_frontend/ArmyFrontend/src/app/Data/Vacancy.ts
import { User } from "./User";
import { VacancyExam } from "./VacancyExam";

export interface Vacancy{
    vacancyId?: string; // Guid is stored as string in JSON
    title: string;
    role: string;
    eligibilityCriteria: string;
    location: string;
    salary: number;
    postedBy: string|null; // Guid as string
    datePosted?: string; // Date as string
    status: string;
    postedByUser?:User;
    vacancyExams?:VacancyExam[];
}

File: Indian_Army_Recruitment/Models/Vacancy.cs
using System.ComponentModel.DataAnnotations;

namespace Indian_Army_Recruitment.Models
{
    public class Vacancy
    {
        [Key]
        public Guid VacancyId { get; set; }

        [Required]
        [StringLength(100)]
        public string Title { get; set; }

        [Required]
        [StringLength(50)]
        public string Role { get; set; }

        [Required]
        public string EligibilityCriteria { get; set; }

        [Required]
        [StringLength(100)]
        public string Location { get; set; }

        [Range(0, double.MaxValue)]
        public int Salary { get; set; }

        public Guid PostedBy { get; set; }
        public DateTime DatePosted { get; set; }

        [Required]
        [StringLength(20)]
        public string Status { get; set; }

        //public User? PostedByUser { get; set; }

        //public ICollection<Application>? Applications { get; set; }
        //public ICollection<VacancyAnalysis>? VacancyAnalysis { get; set; }
        //public ICollection<RequiredDocument>? RequiredDocuments { get; set; }
        //public ICollection<VacancyExam>? VacancyExams { get; set; } 
    }
}
